odl/phantom/noise.py

```diff
@@ -105,6 +108,82 @@ def poisson_noise(intensity):
     return intensity.space.element(values)
 
 
+def salt_pepper_noise(vector, amount=0.05, salt_vs_pepper=0.5,
+                      low_val=None, high_val=None):
+    """Add salt and pepper noise to vector.
+
+    Salt and pepper noise replaces random elements in ``vector`` with
+    ``low_val`` or ``high_val``.
+
+    Parameters
+    ----------
+    vector : `FnBase` or `ProductSpace`
+        The vector that noise should be added to.
+    amount : float, optional
+        The propotion of elements in vector whose element should be converted
+        to noise.
+    salt_vs_pepper : float, optional
+        Relative aboundance of salt (high) vs pepper (low) noise. A high value
+        means more salt than pepper noise.
+    low_val : float, optional
+        The "pepper" color in the noise.
+        Default: minimum value of ``vector``. For product spaces, takes min for
+        each sub-space.
```

Comment:
the minimum value per subspace is taken

If you write "for each", it could be mistaken for "across all"